### PR TITLE
Fix panic in unique persistable channel queue

### LINKED ISSUE #16071: infer panic in queue/unique_queue_disk_channel.go
```
panic: interface conversion: queue.Queue is *queue.LevelUniqueQueue, not *queue.LevelQueue

goroutine 222 [running]:
code.gitea.io/gitea/modules/queue.(*PersistableChannelUniqueQueue).Run.func4(0xc002c487e0)
        /home/lunny/shares/data/Projects/gitea/gitea/modules/queue/unique_queue_disk_channel.go:191 +0x24d
created by code.gitea.io/gitea/modules/queue.(*PersistableChannelUniqueQueue).Run
        /home/lunny/shares/data/Projects/gitea/gitea/modules/queue/unique_queue_disk_channel.go:189 +0x290
```

### PR DESCRIPTION
**Pull request by zeripath**
There is an incorrect casting in the wrapped queue.

Fix #16071

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: modules/queue/unique_queue_disk_channel.go [hunk 1]
```diff
@@ -188,7 +188,7 @@ func (q *PersistableChannelUniqueQueue) Run(atShutdown, atTerminate func(func())
 		go q.internal.Run(func(_ func()) {}, func(_ func()) {})
 		go func() {
 			_ = q.internal.Flush(0)
-			log.Debug("LevelUniqueQueue: %s flushed so shutting down", q.internal.(*LevelQueue).Name())
```
Comment (lunny) [inline modules/queue/unique_queue_disk_channel.go:191]: I want to know if  line 197 also may panic?

Comment (zeripath) [reply modules/queue/unique_queue_disk_channel.go:191]: nope it can't

Comment (lunny) [reply modules/queue/unique_queue_disk_channel.go:191]: But it also `q.internal.(*LevelUniqueQueue)` when `luq, ok := q.internal.(*LevelUniqueQueue); !ok`

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
look's like a copy paste issue ;)

File: modules/queue/unique_queue_disk_channel.go (hunk 1 above)
Comment (zeripath) [reply modules/queue/unique_queue_disk_channel.go:191]: it's only assigned to LevelUniqueQueue.

Comment (lunny) [reply modules/queue/unique_queue_disk_channel.go:191]: so maybe `if q.internal.(*LevelUniqueQueue).ByteFIFOUniqueQueue.byteFIFO.Len(luq.shutdownCtx) != 0` will not make confusion.

**Pull request merged**